File: solutions/python/rna-transcription/1/rna_transcription.py
import string


def to_rna(dna_strand: str) -> str:
    """
    Convert a DNA strand to RNA by replacing G->C, C->G, T->A, A->U.

    Args:
        dna_strand: The DNA strand to convert.

    Returns:
        The RNA strand corresponding to the input DNA strand.
    """
    encoding_table = str.maketrans("GCTA", "CGAU")

    return dna_strand.translate(encoding_table)

File: solutions/typescript/rna-transcription/1/rna-transcription.ts
const dnaToRnaMap: Record<string, string> = {
  G: "C",
  C: "G",
  T: "A",
  A: "U",
} as const;

export function toRna(dna: string): string {
  let rna = "";
  for (const base of dna) {
    const mapped = dnaToRnaMap[base];
    if (!mapped) {
      throw new Error("Invalid input DNA.");
    }
    rna += mapped;
  }
  return rna;
}
